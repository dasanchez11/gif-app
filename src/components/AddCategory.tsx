import { ChangeEvent, FormEvent, useState } from "react";

export interface AddCategoryProps {
  onNewCategory: Function;
}

export const AddCategory = ({ onNewCategory }: AddCategoryProps) => {
  const [inputValue, setInputValue] = useState<string>("");

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const value = inputValue.trim();
    if (value.length <= 1) return;
    onNewCategory(value);
    setInputValue("");
  };

  return (
    <form action="" onSubmit={handleSubmit} aria-label="form">
      <input
        value={inputValue}
        onChange={handleChange}
        type="text"
        placeholder="Search gifs and tap enter"
      />
    </form>
  );
};
