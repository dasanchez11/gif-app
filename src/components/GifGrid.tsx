import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }: { category: string }) => {
  const { images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {loading && <h2>Loading...</h2>}
        {!loading &&
          images.map((image) => <GifGridItem key={image.id} {...image} />)}
      </div>
    </>
  );
};
