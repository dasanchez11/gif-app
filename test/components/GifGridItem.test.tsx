import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";
import React from "react";

describe("Test <GifGridItem/>", () => {
  let title;
  let url;

  beforeEach(() => {
    title = "itemTitle";
    url = "urlString";
  });

  test("should match snapshot", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Should show image with url and alt", () => {
    const title = "itemTitle";
    const url = "urlString";
    render(<GifGridItem title={title} url={url} />);
    const image = screen.getByRole("img");
    const src = image.getAttribute("src");
    const alt = image.getAttribute("alt");
    expect(src).toEqual(url);
    expect(alt).toEqual(title);
  });

  test("should show title", () => {
    render(<GifGridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
