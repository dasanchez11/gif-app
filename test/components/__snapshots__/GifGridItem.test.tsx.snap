// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem/> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="itemTitle"
      src="urlString"
    />
    <p>
      itemTitle
    </p>
  </div>
</div>
`;
